### PR TITLE
HPCC-17144 Strange code in getScopePermissions

### PR DESCRIPTION
**Pull request by RussWhitehead**
getScopePermissions is allowing callers to pass in an invalid auditflags
argument which callers use to determine whether or not permissions are
available (security enabled).  Callers use this hack to first determine
if they should perform security checks.  This PR removes that peculiar
behavior and exposes the once private "permissionsavail" flag, which allows
callers to perform this check in a more intuitive manner

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by RussWhitehead** [comment]
@rpastrana Please review

File: dali/base/dadfs.cpp [hunk 1]
```diff
@@ -1243,13 +1243,11 @@ static void setUserDescriptor(Linked<IUserDescriptor> &udesc,IUserDescriptor *us
     udesc.set(user);
 }
 
+static bool scopePermissionsAvail = true;
```
Comment (rpastrana) [inline dali/base/dadfs.cpp]: This bool is now publicly accessible. Consider encapsulating it along with a getter and the getScopePermissions method.

**Review by rpastrana — COMMENTED** [review]
@RussWhitehead these changes look good, take a look at my one comment which is easily implemented.

File: dali/base/dadfs.cpp (hunk 1 above)
Comment (RussWhitehead) [inline dali/base/dadfs.cpp]: Although I like the idea of disallowing others from changing this variable, I think it's beyond the scope of this PR.   Also, the pattern throughout this C style file is to declare them static with file scope

Comment (rpastrana) [inline dali/base/dadfs.cpp]: Agreed with the comment on the preexisting C-style pattern throughout the file...

**Comment by RussWhitehead** [comment]
@ghalliday Please merge

**Pull request merged**